### PR TITLE
feat(dlp): added a sample for dlp create job

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description

- Added a sample for [dlp create job](https://cloud.google.com/dlp/docs/samples/dlp-create-job) with a unit test case.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by telpirion — APPROVED** [review]

**Comment by telpirion** [comment]
It seems like there are issues with the bucket used for the tests. Please fix.

**Comment by AarshD-crest** [comment]
> It seems like there are issues with the bucket used for the tests. Please fix.

Thank you @telpirion for helping out, There is an issue with bucket creation, Fixed now.

**Review by annadata — APPROVED** [review]

File: dlp/snippets/jobs/jobs_test.go [hunk 1]
```diff
@@ -182,6 +186,133 @@ func TestDeleteJob(t *testing.T) {
...
+func TestCreateJob(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	var buf bytes.Buffer
+	// createBucketForCreatJob will create a bucket and upload a txt file
+	bucketName, err := createBucketForCreatJob(t, tc.ProjectID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	gcsPath := "gs://dlp-go-lang-test/test.txt"
```
Comment (annadata) [inline dlp/snippets/jobs/jobs_test.go:198]: Tests are failing. Looks like gcsPath is hardcoded and should be based on the bucket just created.

Comment (AarshD-crest) [reply dlp/snippets/jobs/jobs_test.go:198]: - Not sure but as per the error in CI Looks like it's permission related issue, https://source.cloud.google.com/results/invocations/4dbc0e64-6595-4177-ab8b-b0ecbbfcc9cf/targets/github%2Fgolang-samples%2Fdlp%2Fsnippets%2Fjobs/tests 
Still looking at it.

Comment (AarshD-crest) [reply dlp/snippets/jobs/jobs_test.go]: Fixed it.

**Pull request merged**